src/audio-manager.ts

```diff
@@ -0,0 +1,248 @@
+                loop: this.loop,
+            });
+            if (posVec !== undefined) {
+                this._pannerNode = new PannerNode(_audioContext, {
+                    coneInnerAngle: 360,
+                    coneOuterAngle: 0,
+                    coneOuterGain: 0,
+                    distanceModel: 'exponential' as DistanceModelType,
+                    maxDistance: 10000,
+                    refDistance: 1.0,
+                    rolloffFactor: 1.0,
+                    panningModel: this.HRTF ? 'HRTF' : 'equalpower',
+                    positionX: posVec[0],
+                    positionY: posVec[2],
+                    positionZ: -posVec[1],
+                    orientationX: 0,
+                    orientationY: 0,
+                    orientationZ: 1,
+                });
+                this._audioNode.connect(this._pannerNode).connect(this._gainNode);
+            } else {
+                this._audioNode.connect(this._gainNode);
+            }
+            this._audioNode.addEventListener('ended', this.stop);
+            this._audioNode.start();
+            this._isPlaying = true;
+        } catch (e) {
+            console.warn(e);
+        }
+    }
+
+    /**
+     * Stops the playback, and if set to destroy, removes associated audio file.
+     */
+    stop() {
+        if (this.isPlaying) {
+            this._audioNode.removeEventListener('ended', this.stop);
+            this._audioNode.stop();
+        }
+        if (this._audioNode !== undefined) {
+            this._audioNode.disconnect();
+        }
+        if (this._pannerNode !== undefined) {
+            this._pannerNode.disconnect();
+        }
+        this._isPlaying = false;
+        if (this._destroy) {
+            this._audioManager._remove(this._source);
+            this._gainNode.disconnect();
+        }
+    }
+
+    /**
+     * Checks if the audio node is currently playing.
+     */
+    get isPlaying(): boolean {
+        return this._isPlaying;
+    }
+
+    /**
+     * Sets the volume of this PlayableNode.
+     */
+    set volume(v: number) {
+        const time = _audioContext.currentTime + RAMP_TIME;
+        this._gainNode.gain.exponentialRampToValueAtTime(v, time);
+    }
+
+    /**
+     * Free's up the audio resources after Node stopped playing.
+     *
+     * @example
+     * ```js
+     * this.audio.play() // plays entire audio file
+     * this.destroy()    // frees resources
+     * this.audio.play() // does nothing
+     * ```
+     */
+    destroy() {
+        if (this._isPlaying) {
+            this._destroy = true;
+        } else {
+            this._audioManager._remove(this._source);
+            this._gainNode.disconnect();
+        }
+
+        /* Remove ability to re-trigger the sound */
+        this.play = this._removePlay.bind(this);
+        this.destroy = () => {};
+    }
+
+    private async _removePlay(): Promise<void> {}
+}
+
+export const globalAudioManager = new AudioManager();
```

Comment:
Since it's a global, maybe `GlobalAudioManager`? Not sure if we ever settled that rule @Squareys

Reply:
We didn't, but `GlobalAudioManager` looks like a type to me 🤔 I would prefer camelCase.